crates/ruff_server/src/server.rs

```diff
@@ -66,10 +66,19 @@ impl Server {
 
         crate::message::init_messenger(connection.make_sender());
 
+        tracing::info!(
+            "Initialization options: {:?}",
+            init_params.initialization_options
+        );
+
         let AllSettings {
             global_settings,
             mut workspace_settings,
-        } = AllSettings::from_value(init_params.initialization_options.unwrap_or_default());
+        } = AllSettings::from_value(
+            init_params
+                .initialization_options
+                .unwrap_or_else(|| serde_json::Value::Object(serde_json::Map::default())),
```

Comment:
What's the difference between this and the default?

Reply:
The default `serde_json::Value` is `Null`, which we cannot successfully deserialize `InitializationOptions` from. However, we can successfully deserialize `InitalizationOptions` from an empty object (`{}`), since all client settings fields are optional.